### PR TITLE
add no_empty_character_class rule

### PR DESCRIPTION
**Pull request by disizali**
this PR is going to add :
+ `no empty character class rule`
+ `3 tests about it`

### CODE REVIEW

File: src/rules/no_empty_character_class.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  pub fn new(context: Context) -> Self {
+    Self { context }
+  }
+}
+
+impl Visit for NoEmptyCharacterClassVisitor {
+  fn visit_regex(&mut self, regex: &Regex, _parent: &dyn Node) {
+    let regex_literal = format!("/{}/", regex.exp);
+    let rule_regex =
+      regex::Regex::new(r"^/([^\[]|\.|\[([^\\\]]|\\.)+\])*/[gimuys]*$")
```
Comment (bartlomieju) [inline src/rules/no_empty_character_class.rs:39]: make this regex a `lazy_static` and please add comment describing this regex as well as linking to eslint rule

File: src/rules/no_empty_character_class.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use crate::test_util::*;
+
+  #[test]
+  fn no_empty_character_class_match() {
+    assert_lint_err::<NoEmptyCharacterClass>(r#"/^abc[]/.test("abcdefg");"#, 0);
```
Comment (bartlomieju) [inline src/rules/no_empty_character_class.rs:58]: Can you port some more test cases from eslint?

Comment (disizali) [reply src/rules/no_empty_character_class.rs:85]: done

File: src/rules/no_empty_character_class.rs (hunk 1 above)
Comment (disizali) [reply src/rules/no_empty_character_class.rs]: thanks for the notice , done!

**Pull request merged**